velox/expression/Expr.cpp

```diff
@@ -669,19 +707,19 @@ void Expr::eval(
     const SelectivityVector& rows,
     EvalCtx& context,
     VectorPtr& result,
-    bool topLevel) {
+    const ExprSet* parentExprSet) {
   if (supportsFlatNoNullsFastPath_ && context.throwOnError() &&
       context.inputFlatNoNulls() && rows.countSelected() < 1'000) {
-    evalFlatNoNulls(rows, context, result, topLevel);
+    evalFlatNoNulls(rows, context, result, parentExprSet);
     return;
   }
 
   // Make sure to include current expression in the error message in case of an
   // exception.
-  ExprExceptionContext exprExceptionContext{this, context.row()};
+  ExprExceptionContext exprExceptionContext{this, context.row(), parentExprSet};
   ExceptionContextSetter exceptionContext(
-      {topLevel ? onTopLevelException : onException,
-       topLevel ? (void*)&exprExceptionContext : this});
+      {parentExprSet ? onTopLevelException : onException,
```

Comment:
nit: wonder if we should change name of onTopLevelException.

Reply:
good point, I tried to associate setting parentExprSet with it being a expression being a top level expression but was not able to reconcile the naming to imply association. Any suggestions?